### PR TITLE
[tests] Allow ConfigurationSchema.json tests to run out of repo

### PR DESCRIPTION
**Pull request by radical**
#### [tests] ConformanceTests: Remove dependency on GetRepoRoot
    .. by changing the tests to expect `ConfigurationSchema.json` to be
    available next to the test assembly.
####  [tests] Avoid redundant tests in Redis Caching tests

    Both `Aspire.StackExchange.Redis.DistributedCaching`, and
    `Aspire.StackExchange.Redis.OutputCaching` have
    `ConfigurationSchema.json` which reference `Aspire.StackExchange.Redis`
    without adding anything more.

    But the tests for those two run validation for the
    `Aspire.StackExchange.Redis` schema, which is already done by the main
    tests. Instead of that:

    - `Aspire.StackExchange.Redis` - continue doing the validation tests
    - `Aspire.StackExchange.Redis.{Output,Distributed}Caching`
        - Run the validation test for the schema against the the project's
          own schema
        - And don't check for any invalid cases since there is nothing being
          added in the schema
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/1790)

### CODE REVIEW

File: tests/Aspire.Azure.AI.OpenAI.Tests/Aspire.Azure.AI.OpenAI.Tests.csproj [hunk 1]
```diff
@@ -5,6 +5,8 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <None Include="$(RepoRoot)src/Components/Aspire.Azure.AI.OpenAI/ConfigurationSchema.json" CopyToOutputDirectory="PreserveNewest" />
```
Comment (eerhardt) [inline tests/Aspire.Azure.AI.OpenAI.Tests/Aspire.Azure.AI.OpenAI.Tests.csproj:8]: (nit) can we use consistent slashes in the .csproj? Everywhere else uses `\`, see the 2 lines below.

File: tests/Aspire.StackExchange.Redis.DistributedCaching.Tests/Aspire.StackExchange.Redis.DistributedCaching.Tests.csproj [hunk 2]
```diff
@@ -5,6 +5,8 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <None Include="$(RepoRoot)src\Components\Aspire.StackExchange.Redis.DistributedCaching\ConfigurationSchema.json" CopyToOutputDirectory="PreserveNewest" />
```
Comment (eerhardt) [inline tests/Aspire.StackExchange.Redis.DistributedCaching.Tests/Aspire.StackExchange.Redis.DistributedCaching.Tests.csproj:8]: Why do we need to copy this file for the DistributedCaching and OutputCaching tests, if the file isn't going to be used?

I like that the base class will fail the test if the file can't be found. But it feels like there should be an "opt out" mechanism that these 2 tests can use to say "skip the ConfigurationSchema tests". Maybe a virtual boolean property on the base class that gets set in these 2 derived classes.

**Review by eerhardt — APPROVED** [review]
Thanks for fixing these.

Looks good. I just have one suggestion and a nit.

File: tests/Aspire.StackExchange.Redis.DistributedCaching.Tests/Aspire.StackExchange.Redis.DistributedCaching.Tests.csproj (hunk 2 above)
Comment (radical) [reply tests/Aspire.StackExchange.Redis.DistributedCaching.Tests/Aspire.StackExchange.Redis.DistributedCaching.Tests.csproj:8]: > Why do we need to copy this file for the DistributedCaching and OutputCaching tests, if the file isn't going to be used?

My thinking was that we will at least be running `JsonSchema.FromFile` which would hopefully validate that the file is a valid xml/schema file.

Comment (eerhardt) [reply tests/Aspire.StackExchange.Redis.DistributedCaching.Tests/Aspire.StackExchange.Redis.DistributedCaching.Tests.csproj:8]: Ok, this makes sense. thanks.

**Review by eerhardt — APPROVED** [review]

**Pull request merged**